smdebug/tensorflow/keras.py

```diff
@@ -479,14 +518,15 @@ def _on_any_batch_begin(self, batch, mode, logs=None):
             self._prepare_collections()
 
         if self._prepared_tensors[mode] is False:
-            if self._validate_exec_function(self._get_exec_function(mode)):
+            if version.parse(tf.__version__) >= version.parse(
+                "2.0.0"
+            ) or self._validate_exec_function(self._get_exec_function(mode)):
                 self._prepare_layers(mode)
                 self._prepare_non_layer_tensors()
                 self._prepared_tensors[mode] = True
                 # below should be after tensors are processed,
                 # so we know that device map is populated
                 self._set_chief_worker()
-            # else:
```

Comment:
Removing this can cause the comment to be misconstrued. Can you modify it as appropriate